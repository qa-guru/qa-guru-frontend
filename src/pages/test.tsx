import { FC } from "react";
import { useParams } from "react-router-dom";

import TestContainer from "features/test/containers/test-container";

const TestPage: FC = () => {
  const { testId, trainingId, lectureId } = useParams<{
    testId: string;
    trainingId: string;
    lectureId: string;
  }>();

  // Проверяем, что все параметры присутствуют
  if (!testId || !trainingId || !lectureId) {
    return <div>Ошибка: отсутствуют необходимые параметры</div>;
  }

  return (
    <TestContainer
      testId={testId}
      trainingId={trainingId}
      lectureId={lectureId}
    />
  );
};

export default TestPage;
