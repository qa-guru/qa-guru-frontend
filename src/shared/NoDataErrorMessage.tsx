import React, { useEffect } from "react";
import { useModal } from "react-modal-hook";
import { Backdrop, Box, Dialog, DialogTitle } from "@mui/material";
import ReplayIcon from "@mui/icons-material/Replay";

const style = {
  icon: { display: "flex", justifyContent: "center", pb: "20px" },
};

const NoDataErrorMessage: React.FC = () => {
  const [showModal] = useModal(({ in: open }) => (
    <Dialog open={open}>
      <DialogTitle>Упс что-то пошло не так....</DialogTitle>
      <Box sx={style.icon}>
        <ReplayIcon cursor="pointer" onClick={reloadPage} />
      </Box>
    </Dialog>
  ));

  const reloadPage = () => {
    window.location.reload();
  };

  useEffect(() => {
    showModal();
  }, [showModal]);

  return (
    <Backdrop
      sx={{ zIndex: (theme) => theme.zIndex.drawer + 1 }}
      open={true}
    ></Backdrop>
  );
};

export default NoDataErrorMessage;
