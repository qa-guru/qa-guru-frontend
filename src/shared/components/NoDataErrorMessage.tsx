import React, { useEffect } from "react";
import { useModal } from "react-modal-hook";
import { Backdrop, Box, Dialog, DialogTitle } from "@mui/material";
import ReplayIcon from "@mui/icons-material/Replay";
import { style } from "../styles";

const NoDataErrorMessage: React.FC = () => {
  const [showModal] = useModal(({ in: open }) => (
    <Dialog open={open}>
      <DialogTitle>Упс что-то пошло не так....</DialogTitle>
      <Box sx={style.icon}>
        <ReplayIcon cursor="pointer" onClick={reloadPage} />
      </Box>
    </Dialog>
  ));

  const reloadPage = () => {
    window.location.reload();
  };

  useEffect(() => {
    showModal();
  }, [showModal]);

  return <Backdrop sx={style.backdropError} open={true} />;
};

export default NoDataErrorMessage;
