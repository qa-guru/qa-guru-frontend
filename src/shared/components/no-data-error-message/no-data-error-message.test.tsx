import { MemoryRouter } from "react-router-dom";
import NoDataErrorMessage from "./no-data-error-message";
import { render } from "../../../test/utils-test";

describe("NoDataErrorMessage", () => {
  it("renders correctly", () => {
    const { asFragment } = render(
      <MemoryRouter>
        <NoDataErrorMessage />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
