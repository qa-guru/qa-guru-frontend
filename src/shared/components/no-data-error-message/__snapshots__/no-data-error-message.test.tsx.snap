// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NoDataErrorMessage > renders correctly 1`] = `
<DocumentFragment>
  <div
    aria-hidden="true"
    class="MuiBackdrop-root css-1q8n7u2-MuiBackdrop-root"
    style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
  />
</DocumentFragment>
`;
